transformers/component_tests/queries/test_helpers/test_helpers.go

```diff
@@ -253,10 +253,10 @@ func CreateIlk(db *postgres.DB, header core.Header, valuesMap map[string]string,
 	catRepo.SetDB(db)
 	jugRepo.SetDB(db)
 	spotRepo.SetDB(db)
-	CreateVatRecords(header, valuesMap, vatMetadatas, vatRepo)
-	CreateCatRecords(header, valuesMap, catMetadatas, catRepo)
-	CreateJugRecords(header, valuesMap, jugMetadatas, jugRepo)
-	CreateSpotRecords(header, valuesMap, spotMetadatas, spotRepo)
+	CreateVatRecords(diffID, header, valuesMap, vatMetadatas, vatRepo)
+	CreateCatRecords(diffID, header.Id, valuesMap, catMetadatas, catRepo)
+	CreateJugRecords(diffID, header, valuesMap, jugMetadatas, jugRepo)
+	CreateSpotRecords(diffID, header, valuesMap, spotMetadatas, spotRepo)
```

Comment:
If we're going to change the arg from `header` => `header.Id`, I'd be down to apply it to each helper for consistency. I definitely see the rationale for doing so - it's nice not to pass unnecessarily large chunks of data if we only care about that one field. Though I do wonder if it might be advantageous to call `test_helpers.CreateDiffWithHeader` internally to these helpers and then strip it out of the component tests' setups 🤔

Reply:
really good call! updated in f9a8617